scripts/scr_marine_struct/scr_marine_struct.gml

```diff
@@ -145,77 +267,161 @@ global.base_stats = { //tempory stats subject to change by anyone that wishes to
 			technology :3,
 			luck :4,
 			skills: {},	
-			gear:{"armour":"Power Armour", "wep1":"Power Sword", "wep2":"Storm Shield"}
+			gear:{"armour":"Power Armour", "wep1":"Power Sword", "wep2":"Storm Shield"},
+			base_group : "human",
 	},
 	"ork_sniper":{
 			title : "Ork Sniper",
-			strength:5,
-			constitution:9,
-			dexterity:8,
-			intelligence:2,
-			wisdom:3,
-			charisma :4,
+			strength:24,
+			constitution:29,
+			dexterity:30,
+			intelligence:10,
+			wisdom:20,
+			charisma :25,
 			religion : "gorkamorka",
 			piety : 20,
-			technology :2,
-			luck :4,
+			technology :8,
+			luck :6,
 			skills: {},	
-			gear:{"armour":"Power Armour", "wep1":"Power Sword", "wep2":"Storm Shield"}
+			gear:{"armour":"Power Armour", "wep1":"Power Sword", "wep2":"Storm Shield"},
+			base_group : "ork",
 	},
 	"flash_git":{
 			title : "Flash Git",
-			strength:8,
-			constitution:13,
-			dexterity:6,
-			intelligence:2,
+			strength:30,
+			constitution:23,
+			dexterity:15,
+			intelligence:5,
 			wisdom:3,
 			charisma :4,
 			religion : "gorkamorka",
 			piety : 20,
 			technology :2,
 			luck :4,
 			skills: {},	
-			gear:{"armour":"Power Armour", "wep1":"Power Sword", "wep2":"Storm Shield"}
+			gear:{"armour":"Power Armour", "wep1":"Power Sword", "wep2":"Storm Shield"},
+			base_group : "ork",
 	}
 }
 function TTRPG_stats(faction, comp, mar, class = "marine") constructor{
-	
+	constitution=0; strength=0;luck=0;dexterity=0;wisdom=0;piety=0;charisma=0;technology=0;intelligence=0;weapon_skill=0;ballistic_skill=0;size = 0;
+	company = comp;			//marine company
+	marine_number = mar;			//marine number in company
+	obj_ini.bio[company,marine_number] = 0;   //need to rework init of 2d arrays( and eventually remove)
+	static bionics = function(){return obj_ini.bio[company,marine_number];}// get marine bionics count	
+	static experience =  function(){return obj_ini.experience[company,marine_number];}//get exp
+	static update_exp = function(new_val){obj_ini.experience[company,marine_number] =new_val}//change exp
+	static add_exp = function(add_val){obj_ini.experience[company,marine_number] +=add_val}
+	static armour = function(){ 
+		return obj_ini.armour[company,marine_number];
+	};
+	static role = function(){
+		return obj_ini.role[company,marine_number];
+	};
+	static update_role = function(new_role){
+		obj_ini.role[company,marine_number]= new_role;
+		array_append(role_history ,[obj_ini.wep1[company,marine_number], obj_controller.turn])
+	};	
+	static mobility_item = function(){ 
+		return obj_ini.mobi[company,marine_number];
+	};	
+	static hp = function(){ 
+		return obj_ini.hp[company,marine_number]; //return current health
+	};
+    static update_health = function(new_health){
+      obj_ini.hp[company,marine_number] = new_health;
+   };	
+	static get_unit_size = function(){
+		var r = role();
```

Comment:
please use full, normal names for variables, it makes it harder to read than necessary. If you get naming conflicts remember that you can prefix with _

Reply:
will do